tests/factories/cht/contacts/cht_hierarchy.js

```diff
@@ -0,0 +1,58 @@
+const Factory = require('rosie').Factory;
+const uuid = require('uuid');
+
+require('../users/offline');
+require('./district-hospital');
+require('./health-center');
+require('./clinic');
+require('./person');
+
+const districtId = uuid.v4();
+const healthCenterID = uuid.v4();
+const clinicId = uuid.v4();
+const personId = uuid.v4();
```

Comment:
All of these will still only be evaluated once and every time someone uses the factory to generate a new object from here, that new object will have the same id. The way to solve that is to use Rosie's ability to execute a function for a given attribute.

...basically, please change this stuff to work like you have in `tests/factories/cht/contacts/clinic.js` for the `_id` field stuff.

Reply:
@craig-landry  ok have a look again. The idea we talked about got thrown out because there's no way to have rosie build a list and then known about the other objects in that list.  I've create a generic place and then a generate Hierarchy function that generates a small hierarchy based on the types passed in. I'm sure there is room to improve for say when you need multiple places at each level for I'm not sure the tests that need that yet.

Reply:
I think that is a reasonable compromise to handle what we need. We can take another shot of getting fancy another time when we have more use cases to take from.

Regarding the specific code in the implementation, please adjust to approach to first build the objects and then stitch them together into a hierarchy. With the code as it is now I worry that it'll be easy for someone to break over time and at a minimum, it really takes some careful reading to see that it's working.